Rename EntityExtractorMixin (#11356)
RasaHQ/rasa: changelog/10225.improvement.md

> Rename `EntityExtractorMixin` to `EntityExtractor` due to consistency, as it was discussed in issue 10225.

Reviewer: ```suggestion
Rename `EntityExtractorMixin` to `EntityExtractor` for consistency, as was discussed in issue 10225.
```
a nitpick =)
